src/components/organisms/SchedulePageModal/SchedulePageModal.tsx

```diff
@@ -1,162 +1,126 @@
+  const { userWithId } = useUser();
+  const userEventIds =
+    userWithId?.myPersonalizedSchedule ?? emptyPersonalizedSchedule;
+
+  const { relatedVenueEvents, relatedVenues } = useConnectRelatedVenues({
     venueId,
     withEvents: true,
   });
 
-  const relatedVenuesById: Partial<
-    Record<string, WithId<AnyVenue>>
-  > = relatedVenues.reduce(itemsToObjectByIdReducer, {});
+  const [relatedRooms, setRelatedRooms] = useState<Room[]>([]);
+
+  useEffect(() => {
+    const rooms: Room[] = [];
+    relatedVenues
+      .map((venue) => venue.rooms || [])
+      .forEach((venueRooms) => rooms.push(...venueRooms));
+    setRelatedRooms(rooms);
+  }, [relatedVenues]);
 
-  const orderedEvents: DatedEvents = useMemo(() => {
+  const schedule: ScheduleDay[] = useMemo(() => {
     const liveAndFutureEvents = relatedVenueEvents.filter(isEventLiveOrFuture);
-    const hasEvents = liveAndFutureEvents.length > 0;
-
-    const nowDay = startOfDay(new Date());
-
-    const dates: DatedEvents = range(0, DAYS_AHEAD).map((idx) => {
-      const day = addDays(nowDay, idx);
-
-      const todaysEvents = liveAndFutureEvents
-        .filter((event) => {
-          return isWithinInterval(day, {
-            start: startOfDay(new Date(event.start_utc_seconds * 1000)),
-            end: endOfDay(
-              new Date(
-                (event.start_utc_seconds + event.duration_minutes * 60) * 1000
-              )
-            ),
-          });
-        })
-        .sort((a, b) => a.start_utc_seconds - b.start_utc_seconds);
-
-      return {
-        dateDay: day,
-        events: hasEvents ? todaysEvents : [],
-      };
-    });
-
-    return dates;
-  }, [relatedVenueEvents]);
-
-  const [date, setDate] = useState(0);
-
-  const scheduleTabs = useMemo(
+    const today = startOfDay(Date.now());
+    const buildScheduleEvent = scheduleDayBuilder(
+      today,
+      liveAndFutureEvents,
+      relatedRooms,
+      userEventIds
+    );
+
+    return range(0, DAYS_AHEAD).map((dayIndex) => buildScheduleEvent(dayIndex));
+  }, [relatedVenueEvents, relatedRooms, userEventIds]);
+
+  const [selectedDayIndex, setSelectedDayIndex] = useState(0);
+
+  const onWeekdayClick = useCallback(
+    (index: number): MouseEventHandler<HTMLLIElement> => (e) => {
+      e.stopPropagation();
+      setSelectedDayIndex(index);
+    },
+    []
+  );
+
+  const weekdayClasses = useCallback(
+    (index) =>
+      classNames("SchedulePageModal__weekday", {
+        "SchedulePageModal__weekday--active": index === selectedDayIndex,
+      }),
+    [selectedDayIndex]
+  );
+
+  const weekdays = useMemo(
     () =>
-      orderedEvents.map((day, idx) => (
+      schedule.map((day, index) => (
         <li
-          key={formatDate(day.dateDay.getTime())}
-          className={`button ${idx === date ? "active" : ""}`}
-          onClick={(e) => {
-            e.stopPropagation();
-            setDate(idx);
-          }}
+          key={formatDate(day.dayStartUtcSeconds)}
```

Comment:
Probably don't need to call `formatDate` on this, the raw value should also be unique I would imagine?

Reply:
yeah, you're right. replaced it here: https://github.com/sparkletown/sparkle/pull/1302/commits/6c79ed94ad0acdbf8061bba26c4e73f7912ed2de

but then I changed code a bit, so I get Date object there now, that's why transforming it to isoString in this commit: https://github.com/sparkletown/sparkle/pull/1302/commits/9fce282d69bf3236a3cad6381d0b990072e0c3c2